filc/cmds.go

```diff
@@ -317,11 +322,20 @@ var retrieveFileCmd = &cli.Command{
 			}
 		}
 
-		root, err := cid.Decode(cidStr)
+		noIPFS := cctx.Bool(flagNoIPFS.Name)
+
+		// --datamodel-path-selector implies --no-ipfs
+		if cctx.String(flagDmPathSel.Name) != "" {
+			noIPFS = true
+		}
```

Comment:
```suggestion
		if cctx.String(flagDmPathSel.Name) != "" {
			if cctx.IsSet(flagNoIPFS.Name) && ! noIPFS {
				// error here - user explicitly requested an impossible combo, do not quietly override their choice
			}
			noIPFS = true
		}
```

Reply:
Yes please, change it. I wasn't sufficiently clear on slack, I meant to say:

> auto-set it if the user didn't have a preference, but if a user gave you conflicting options: fail, do not "know better" than them

Reply:
@elijaharita a user is super-unlikely to do this knowingly. **BUT** when this utility becomes a part of a spaghetti `.bash` obnoxiousness, and flags start appearing in places without the ability to reason about the entire pipeline anymore: **that** is where "forcing" like this will strike, and be near-impossible to find and debug.

Therefore: failing on nonsensical option-sets is the only reasonable thing to do.